src/lib/loaders/batchLoader.ts

```diff
@@ -0,0 +1,111 @@
+import DataLoader from "dataloader"
+import { chain, flatten, chunk } from "lodash"
+import config from "config"
+
+const { ENABLE_RESOLVER_BATCHING } = config
+
+/**
+ * This is just used to compare options sent into the dataloader.
+ * If the options are an object this stingifies the object in a way
+ * that they can be grouped as similar.
+ */
+const renderParams = key => {
+  if (typeof key === "string") {
+    return ""
+  }
+  const { id, ...params } = key
+  return Object.entries(params)
+    .map(entry => entry.join("="))
+    .sort()
+    .join("&")
+}
+
+interface GroupKeysResult {
+  id: string[]
+  size: number
+  [key: string]: any
+}
+export const groupKeys = (requestedKeys: string | { id }): GroupKeysResult[] =>
+  (chain(requestedKeys)
+    .groupBy(renderParams)
+    .values()
+    .map(values => chunk(values, 20))
+    .flatten()
+    .map((keys: string[] | { id }[]) => {
+      if (typeof keys[0] === "string") {
+        return { id: keys, size: keys.length }
+      }
+      return {
+        // @ts-ignore
+        ...keys[0],
+        id: keys.map(k => k.id),
+        size: keys.length,
+      }
+    })
+    .value() as unknown) as GroupKeysResult[]
```

Comment:
Tiny nit, this would be a little more introspectable if it has an explicit return vs an illicit return (ignoring `tap`, which folks probably shouldn't be expected to know about)